Add mcxaxx6 arduino gpio and usb support

NeilChen93 opened the pull request:
Add arduino_header node labels to frdm_mcxa366,frdm_mcxa346 and frdm_mcxa266 device tree board definition.
Add usb support for frdm_mcxa366 and frdmmcxa266 boards.

hakehuang commented:
> Hi @hakehuang, According to Mark Wang's comment in #100094, this is an common issue, need him to enable "remote wakeup" support of NXP udc drivers later to fix it. My question is, do I need to wait until he resolves this issue before merging my PR?

No need. this is OK for your PR.

mmahadevan108 commented:
@NeilChen93 , please rebase

NeilChen93 committed:
boards: nxp: frdm_mcxaxx6: add arduino labels

Added arduino_header node labels to FRDM-MCXA366,FRDM-MCXA346 and
FRDM-MCXA266 device tree board definition, allowing compatible
shield boards to be used. Also extend the board YAML file with
related support tags arduino_gpio.

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

NeilChen93 committed:
dts: arm/nxp: Add usb nodes to NXP MCXA366 and MCXA266 dtsi file

Add usb nodes to NXP MCXA366 and MCXA266 dtsi file

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

NeilChen93 committed:
boards: nxp: frdm_mcxaxx6: Support usb for NXP frdm_mcxaxx6 board

Support usb for NXP frdm_mcxa1366 and frdm_mcxa266 board.
Test it using samples/subsys/usb/console and tests/drivers/udc

Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>
